### PR TITLE
osrm-backend: fix build on `x86_64-darwin`

### PR DESCRIPTION
**Pull request by shivaraj-bh**
## Description of changes

Adds `"-faligned-allocation”` to `NIX_CFLAGS_COMPILE` fixing: `error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.13 or newer` on `x86_64-darwin`

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by shivaraj-bh** [comment]
Result of `nixpkgs-review pr 344906` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>osrm-backend</li>
  </ul>
</details>

File: pkgs/servers/osrm-backend/default.nix [hunk 1]
```diff
@@ -53,13 +53,27 @@ stdenv.mkDerivation rec {
...
+    [
+      # Needed with GCC 12
+      "-Wno-error=stringop-overflow"
+      "-Wno-error=uninitialized"
+      # Needed for GCC 13
+      "-Wno-error=array-bounds"
+    ]
+    ++
+      # error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.13 or newer
+      (lib.optionals
```
Comment (erictapen) [inline pkgs/servers/osrm-backend/default.nix:66]: I  think you could use `lib.optional` here?

**Review by erictapen — APPROVED** [review]

File: pkgs/servers/osrm-backend/default.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/servers/osrm-backend/default.nix:66]: It's fine to use `lib.optionals`.

**Comment by erictapen** [comment]
Does somebody know what the ofborg errors for darwin mean?

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 344906`

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>osrm-backend</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
> > Does somebody know what the ofborg errors for darwin mean?
> 
> The NixOS tests cannot be executed on darwin since that is another architecture.

Should gate the tests behind a linux check, if that's the case.

**Comment by shivaraj-bh** [comment]
> Should gate the tests behind a linux check, if that's the case.

Currently it is:
```nix
{
  passthru.tests = {
    inherit (nixosTests) osrm-backend;
  };
}
```

I see most of the other packages using the same norm, should there instead be a `treewide` PR running these tests only on Linux?

**Pull request merged**